irohad/ametsuchi/impl/postgres_block_query.cpp

```diff
@@ -242,11 +242,27 @@ namespace iroha {
         };
     }
 
-    bool PostgresBlockQuery::hasTxWithHash(
+    bool PostgresBlockQuery::hasCommittedTxWithHash(
         const shared_model::crypto::Hash &hash) {
       return getBlockId(hash) != boost::none;
     }
 
+    bool PostgresBlockQuery::hasRejectedTxWithHash(
```

Comment:
Why should we separate hashes storage on two methods? 
Why could we not store both kinds of hashes in the same table with different markers in a separate column?

something like:
```
+-----+----------------+--------------------+-----+
| ... | hash           | committed/rejected | ... |
+-----+----------------+--------------------+-----+
|     | 2ac7b67ca6b76c | committed          |     |
+-----+----------------+--------------------+-----+
|     | 89ac7b867a875a | rejected           |     |
+-----+----------------+--------------------+-----+
```

That approach will let us avoid looking up in two independent sources for the single transaction in the worst case. 
Also, we will have only one method for a transaction look up.